src/pamiq_core/gym/agent.py

```diff
@@ -0,0 +1,104 @@
+from abc import abstractmethod
+from dataclasses import asdict
+from typing import Any, override
+
+from pamiq_core.interaction import Agent
+
+from .types import EnvReset, EnvStep, GymAction, GymObs
+
+
+class GymAgent[O, A](Agent[GymObs[O], GymAction[A]]):
+    """Base class for agents that interact with Gymnasium environments.
+
+    This abstract class provides the interface for agents to handle
+    Gymnasium environment observations and produce actions.
+
+    Set `self.need_reset=True` to reset the environment after current step.
+
+    Example:
+        ```python
+        class MyCartPoleAgent(GymAgent[np.ndarray, int]):
+            def on_reset(self, obs, info):
+                # Return initial action
+                return 0
+
+            def on_step(self, obs, reward, truncated, terminated, info):
+                # Simple policy: move right if pole is tilting right
+                return 1 if obs[2] > 0 else 0
+        ```
+    """
+
+    need_reset: bool = False
+
+    @override
+    def setup(self) -> None:
+        """Initialize the agent and reset the need_reset flag."""
+        super().setup()
+        self.need_reset = False
+
+    @abstractmethod
+    def on_reset(self, obs: O, info: dict[str, Any]) -> A:
+        """Handle environment reset and return initial action.
+
+        Args:
+            obs: Initial observation from the environment
+            info: Additional information from the environment
+
+        Returns:
+            The initial action to take
+        """
+        pass
+
+    @abstractmethod
+    def on_step(
+        self,
+        obs: O,
+        reward: float,
+        terminated: bool,
+        truncated: bool,
+        info: dict[str, Any],
+    ) -> A:
+        """Process a step observation and return next action.
+
+        Args:
+            obs: Current observation from the environment
+            reward: Reward received from the previous action
+            truncated: Whether the episode was truncated
+            terminated: Whether the episode terminated naturally
```

Comment:
naturallyがどういう意味なのか具体的に説明してほしさ
```suggestion
            terminated: Whether the episode terminated naturally
```

Reply:
これは gym環境の終了判定が，打ち切り`truncated`と エピソード終了 `terminated`の二つあるからなんだよね．単に `Whether the episode terminated`だけだと `truncated`と何が違うのか伝わんないかなって．

Reply:
であればこういう感じで途中で打ち切られたか、正常終了したかって内容欲しい
```suggestion
            truncated: Whether the episode was truncated before completion
            terminated: Whether the episode terminated successfully
```